app/controllers/admin/delivery_zones_controller.rb

```diff
@@ -0,0 +1,49 @@
+class Admin::DeliveryZonesController < ApplicationController
+
+    def new
+        @delivery_zone = DeliveryZone.new    
+    end
+
+    def create
+        @delivery_zone = DeliveryZone.new(delivery_zone_params)
+        if @delivery_zone.save
+            redirect_to admin_delivery_zones_url
+        else
+            render 'new'
+        end
+    end
+
+    def index
+        @delivery_zones = DeliveryZone.all.order(:name)
+    end
+  
+    def show
+        @delivery_zone = DeliveryZone.find(params[:id])
+    end
+
+    def edit
+        @delivery_zone = DeliveryZone.find(params[:id])
+        redirect_to admin_restaurants_url if @delivery_zone.nil?
+    end
+
+    def update
+        @delivery_zone = DeliveryZone.find(params[:id])
+        if @delivery_zone.update_attributes(delivery_zone_params)
```

Comment:
I've always wondered why folks use `update_attributes` instead of `update`.. perhaps you could enlighten me 😄

Reply:
lol it's an old habit. I was coding in Rails when ```update``` and ```update_attributes``` would return different things. More specifically ```update_attributes``` returned ```false``` if validation and saved failed and back then ```update``` returned the updated object itself, no matter if validation failed or not. Now, more current versions of Rails has ```update``` return false if validation failed and ```update_attributes``` is just an alias of it.